console.go

```diff
@@ -351,17 +330,43 @@ func (w ConsoleWriter) writePart(buf *bytes.Buffer, evt map[string]interface{},
 
 // orderFields takes an array of field names and an array representing field order
 // and returns an array with any ordered fields at the beginning, in order,
-// and the remaining fields after in their original order.
+// and the remaining fields after in their original order or sorted alphabetically.
 func (w ConsoleWriter) orderFields(fields []string) {
 	if w.fieldIsOrdered == nil {
 		w.fieldIsOrdered = make(map[string]int)
 		for i, fieldName := range w.FieldsOrder {
 			w.fieldIsOrdered[fieldName] = i
 		}
 	}
-	sort.Slice(fields, func(i, j int) bool {
-		ii, iOrdered := w.fieldIsOrdered[fields[i]]
-		jj, jOrdered := w.fieldIsOrdered[fields[j]]
+
+	indexes := w.fieldIsOrdered
+	tiebreaker := func(i, j int) bool {
+		return fields[i] < fields[j]
+	}
+
+	if w.DoNotSortFields {
+		offset := len(w.fieldIsOrdered)
+		indexes = make(map[string]int, offset)
+		for fieldName, index := range w.fieldIsOrdered {
+			indexes[fieldName] = index
+		}
+
+		for i, field := range fields {
+			if _, ok := indexes[field]; !ok {
+				indexes[field] = offset + i
+			}
+		}
+
+		tiebreaker = func(i, j int) bool {
+			panic("unreachable")
+		}
+	}
+
+	indexes[ErrorFieldName] = -1 // Always put error field first
```

Comment:
Force the field matching `ErrorFieldName` to the front. This eliminates the entire rearrangement block in `writeFields`